compiler/bin-wasm_of_ocaml/compile.ml

```diff
@@ -155,19 +165,52 @@ let optimize ~profile in_file out_file =
     ("wasm-opt"
     :: (common_binaryen_options ()
        @ optimization_options.(level - 1)
-       @ [ Filename.quote in_file; "-o"; Filename.quote out_file ]))
+       @ [ Filename.quote in_file; "-o"; Filename.quote out_file ])
+       @ (match sourcemap_file with
+          | Some sourcemap_file ->
+              [ "--input-source-map"
+              ; Filename.quote (in_file ^ ".map")
+              ; "--output-source-map"
+              ; Filename.quote sourcemap_file
+              ; "--output-source-map-url"
+              ; Filename.quote sourcemap_file ]
+          | None -> []))
 
-let link_and_optimize ~profile runtime_wasm_files wat_file output_file =
+let link_and_optimize
+      ~profile
+      ?sourcemap_file
+      runtime_wasm_files
+      wat_file
+      output_file =
+  let sourcemap_file =
+    if Sys.command "wasm-merge -osm foo 2> /dev/null" <> 0 then None
```

Comment:
```suggestion
    if Option.is_none sourcemap_file || Sys.command "wasm-merge -osm foo 2> /dev/null" <> 0 then None
```

Reply:
This would still call `wasm-metadce` with the `-osm` option though.

Reply:
If `sourcemap_file` is `None`, one does not need to check for source map support, and `wasm-metadce` is not going to be called with the `-osm` option, right?

Reply:
Ah yes, I was confused. Done.